### ISSUE TITLE
better display of labels with `ggrepel` in `ggscatterstats`

### ISSUE THREAD

**Issue by IndrajeetPatil** [IndrajeetPatil/ggstatsplot#116]
The segments connecting points to labels don't start at the point. This is not noticeable when there are a lot of points, but is pretty conspicuous with very few data points.
``` r

library(ggstatsplot)

# creating a small dataframe
set.seed(123)
df <- ggstatsplot::movies_long %>%
  dplyr::filter(.data = ., 
                genre %in% c("Comedy", "Drama"), 
                mpaa %in% c("PG", "R")) %>%
  dplyr::sample_frac(tbl = ., size = 0.015)

# plot
ggstatsplot::ggscatterstats(
  data = df,
  x = "length",
  y = rating,
  label.var = title,
  results.subtitle = FALSE,
  messages = FALSE,
  marginal = FALSE
)
```

![](https://i.imgur.com/CiHpW8m.png)

<sup>Created on 2018-12-22 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by ibecav** [comment]
This is because the default for point jitter is 40% and ggrepel doesn't honor point jitter see reprex

``` r
library(ggstatsplot)
# creating a small dataframe

set.seed(123)
df <- ggstatsplot::movies_long %>%
  dplyr::filter(.data = .,
  genre %in% c("Comedy", "Drama"),
  mpaa %in% c("PG", "R")) %>%
  dplyr::sample_frac(tbl = ., size = 0.015)
# plot
ggstatsplot::ggscatterstats(
  data = df,
  x = "length",
  y = rating,
  label.var = title,
  results.subtitle = FALSE,
  messages = FALSE, , 
  point.width.jitter = 0, 
  point.height.jitter = 0,
  marginal = FALSE
)
```

![](https://i.imgur.com/TjZfbzf.png)

<sup>Created on 2018-12-25 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by ibecav** [comment]
Thanks.  just getting back into this after the holiday.  I followed the link and understand the fix.  BUT, I guess my final comment is why have the **_default_** set at 40% when it causes this issue + (and this is actually more important to me) jittering makes any visual inspection inherently less accurate?

Not to be argumentative but to my point, when you look at your original reprex above consider the movie "Edges of the Lord" your default settings leave the viewer confused as to whether the `length` is greater than or less than 95.  Whereas if you make the default `point.width.jitter = 0` the viewer can actually see the answer is almost exactly 95...

**Comment by IndrajeetPatil** [comment]
Good point. I was defaulting to `ggplot2::position_jitter` defaults. But you are right that this is clearly visually misleading! I find this useful when I have ordinal scale and a lot of points overlap (see "(b)" in figure below; esp. when `rating == 1`), but the marginals are pretty misleading in this case.

``` r
library(tidyverse)

# creating a new dataset
set.seed(123)
dat <- ggstatsplot::intent_morality %>%
  dplyr::filter(.data = .,
                id > 1,
                id <= 500,
                harm == "Poisoning",
                condition == "neutral") %>% # creating fake age column
  dplyr::mutate(.data = .,
                age = rnorm(n = nrow(.), mean = 50, sd = 5))

# plot
ggstatsplot::combine_plots(
  ggstatsplot::ggscatterstats(
    data = dat,
    x = age,
    y = rating,
    messages = FALSE,
    results.subtitle = FALSE,
    title = "with defaults"
  ),
  ggstatsplot::ggscatterstats(
    data = dat,
    x = age,
    y = rating,
    messages = FALSE,
    results.subtitle = FALSE,
    point.width.jitter = 0,
    point.height.jitter = 0,
    title = "no jitter"
  ),
  labels = c("(a)", "(b)")
)
```

![](https://i.imgur.com/Eeu6hd2.png)

I will change the defaults so that **there is no jitter**.

That said, this still just postpones this issue which someone is likely to encounter if they jitter points. So better to resolve this sooner than later. 

Do you want to take a look at fixing this issue and then I can change the defaults?

**Comment by ibecav** [comment]
I agree it doesn't solve the jitter issue per se but I do believe the the default of no jitter will let the user make an informed decision.

I'm happy to look at this after the January 5th release date if you like.  The only thing I was going to prioritize before then was the NA issue and refactoring of grouped_ggbetweenstats.  I think I can do both at the same time just not sure it will be all done before the 5th.

**Comment by ibecav** [comment]
I put in the PR to fix #103 I'll look at a refactor next and if I have time swing back to this jitter issue

**Comment by IndrajeetPatil** [comment]
I will modify the defaults for jitter now to set them to 0.
